WooCommerce/Classes/Authentication/PostSiteCredentialLoginChecker.swift

```diff
@@ -0,0 +1,192 @@
+import Yosemite
+import protocol Networking.ApplicationPasswordUseCase
+import enum Networking.ApplicationPasswordUseCaseError
+
+/// Checks if the user is eligible to use the app after logging in with site credentials only.
+/// The following checks are made:
+/// - Application password availability
+/// - Role eligibility
+/// - Whether WooCommerce is installed and activated on the logged in site.
+///
+final class PostSiteCredentialLoginChecker {
+    private let stores: StoresManager
+    private let applicationPasswordUseCase: ApplicationPasswordUseCase
+    private let roleEligibilityUseCase: RoleEligibilityUseCaseProtocol
+
+    init(applicationPasswordUseCase: ApplicationPasswordUseCase,
+         roleEligibilityUseCase: RoleEligibilityUseCaseProtocol = RoleEligibilityUseCase(stores: ServiceLocator.stores),
+         stores: StoresManager = ServiceLocator.stores) {
+        self.applicationPasswordUseCase = applicationPasswordUseCase
+        self.roleEligibilityUseCase = roleEligibilityUseCase
+        self.stores = stores
+    }
+
+    /// Checks whether the user is eligible to use the app.
+    ///
+    func checkEligibility(for siteURL: String, from navigationController: UINavigationController, onSuccess: @escaping () -> Void) {
+        checkApplicationPassword(for: siteURL,
+                                 with: applicationPasswordUseCase,
+                                 in: navigationController) { [weak self] in
+            self?.checkRoleEligibility(in: navigationController) {
+                self?.checkWooInstallation(for: siteURL, in: navigationController, onSuccess: onSuccess)
```

Comment:
❓ Can we send these requests in parallel? Do we have to wait for the role eligibility response before checking Woo installation?

Reply:
I think we want to avoid presenting a few alerts at the same time in case there is more than one issue with the checks. If we want to call everything concurrently, we'll need another mechanism to check the results and prioritize the alerts we want to display. For simplicity, I'm using nested closures for now, I hope it's alright.

Reply:
I agree. 👍